packages/core/src/button/button.stories.ts

```diff
@@ -163,31 +163,33 @@ export const links = () => {
   return html`
     <div cds-layout="horizontal gap:sm">
       <cds-button>
-        link
-      </cds-button>
-      <cds-button>
-        <a href="javascript:void(0)">link</a>
+        button
       </cds-button>
 
-      <cds-button>
-        <a href="javascript:void(0)">this is a long link</a>
-      </cds-button>
+      <a href="javascript:void(0)">
+        <cds-button>link</cds-button>
+      </a>
+
+      <a href="javascript:void(0)">
+        <cds-button>this is a long link</cds-button>
+      </a>
+
+      <a href="javascript:void(0)">
+        <cds-button size="sm">small link</cds-button>
+      </a>
 
-      <cds-button size="sm">
-        <a href="javascript:void(0)">small link</a>
-      </cds-button>
       <br />
-      <cds-button action="outline">
-        <a href="javascript:void(0)">link</a>
-      </cds-button>
+      <a href="javascript:void(0)">
+        <cds-button action="outline">link</cds-button>
+      </a>
 
-      <cds-button action="outline">
-        <a href="javascript:void(0)">this is a long link</a>
-      </cds-button>
+      <a href="javascript:void(0)">
+        <cds-button action="outline">this is a long link</cds-button>
+      </a>
 
-      <cds-button action="outline" size="sm">
-        <a href="javascript:void(0)">small link</a>
-      </cds-button>
+      <a href="javascript:void(0)">
+        <cds-button action="outline" size="sm">small link</cds-button>
+      </a>
```

Comment:
Based on our internal conversations this adds support to wrap buttons with an anchor. This is not a breaking change the slotted anchor behavior remains in 4.0. The base button will detect the anchor and switch to readonly mode preserving the proper semantics and a11y of the link.

The wrapping style API seems to be far more consistent/predictable with styles and easier to support framework behavior with custom routing solutions.